File: 2560-house-robber-iv/2560-house-robber-iv.cpp
class Solution {
public:
    int minCapability(vector<int>& nums, int k) {
        int s=1;
        int e=1e9,ans=1e9;
        while(s<=e){
            int mid=(s+e)/2;
            int counter=0;
            for(int i=0;i<nums.size();i++){
                if(nums[i]<=mid){
                    counter++;
                    i++;
                }
                
            }
            if(counter>=k){
                ans=min(ans,mid);
                e=mid-1;
            }
            else{
                s=mid+1;
            }
        }
   return ans; }
};

File: 2560-house-robber-iv/2560-house-robber-iv.java
class Solution {
    public int minCapability(int[] nums, int k) {
        int left = 1, right = (int) 1e9, ans = (int) 1e9;

        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (canSelect(nums, mid, k)) {
                ans = mid;
                right = mid - 1;
            } else {
                left = mid + 1; 
            }
        }
        return ans;
    }

    private boolean canSelect(int[] nums, int mid, int k) {
        int count = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] <= mid) {
                count++;
                i++; 
            }
        }
        return count >= k;
    }
}
